bin/templates/cordova/Api.js

```diff
@@ -208,17 +208,13 @@ Api.prototype.addPlugin = function (plugin, installOptions) {
         installOptions.variables.PACKAGE_NAME = project.getPackageName();
     }
 
-    if (this.android_studio === true) {
-        installOptions.android_studio = true;
-    }
-
     return Q().then(function () {
         return PluginManager.get(self.platform, self.locations, project).addPlugin(plugin, installOptions);
     }).then(function () {
         if (plugin.getFrameworks(this.platform).length === 0) return;
         selfEvents.emit('verbose', 'Updating build files since android plugin contained <framework>');
         // This should pick the correct builder, not just get gradle
-        require('./lib/builders/builders').getBuilder(this.builder).prepBuildFiles();
+        require('./lib/builders/builders').getBuilder().prepBuildFiles();
```

Comment:
Probably for the next round of cleaning, but there's a bunch of stuff around here that could be simplified with arrow functions and could use native Promises.

Reply:
I also agree. Some of the things I was thinking about in the clean up PR would be:
- arrow functions
- native promises
- `let` and `const`
- Use `fs-extra` to replace things like `shell.cp` etc.
- Remove `builders/builders.js`

Reply:
(Best create a cleanup issue with these tasks so they won't be forgotten)

Reply:
Will do.